src/ln/channelmanager.rs

```diff
@@ -1806,8 +1819,23 @@ impl ChannelManager {
 		};
 		// Because we have exclusive ownership of the channel here we can release the channel_state
 		// lock before add_update_monitor
-		if let Err(_e) = self.monitor.add_update_monitor(monitor_update.get_funding_txo().unwrap(), monitor_update) {
-			unimplemented!();
+		if let Err(e) = self.monitor.add_update_monitor(monitor_update.get_funding_txo().unwrap(), monitor_update) {
+			match e {
+				ChannelMonitorUpdateErr::PermanentFailure => {
+					// Note that we reply with the new channel_id in error messages if we gave up on the
+					// channel, not the temporary_channel_id. This is compatible with ourselves, but the
+					// spec is somewhat ambiguous here. Not a huge deal since we'll send error messages for
+					// any messages referencing a previously-closed channel anyway.
+					return Err(MsgHandleErrInternal::from_finish_shutdown("ChannelMonitor storage failure", funding_msg.channel_id, chan.force_shutdown(), None));
```

Comment:
Not specifically here, but in handle_monitor_err we disclose to our peer than we have a monitor failure. That's kinda overlap with our previous talk on data_loss_protection, IMO still have doubts on this. But you can't reply anyway that a serious attacker able to fingerprint a rust-lightning node can know our error call flow so even if we tell another content he will know that's a monitor failure...

Reply:
Right, but in theory the point of a monitor error is to close ASAP because we still have a monitor in memory that is fully up-to-date. Its only if the node crashes that we're screwed.